executor/index_lookup_hash_join.go

```diff
@@ -305,9 +431,12 @@ func (iw *indexHashJoinInnerWorker) run(ctx context.Context, cancelFunc context.
 		logutil.Logger(ctx).Error("indexHashJoinInnerWorker.run failed", zap.Error(joinResult.err))
 		return
 	}
-	if joinResult.chk != nil && joinResult.chk.NumRows() > 0 {
+	// When task.keepOuterOrder is TRUE(resultCh != iw.resultCh), the last
+	// joinResult will be checked when the a task has been processed, thus we do
+	// not need to check it here again.
+	if resultCh == iw.resultCh && joinResult.chk != nil && joinResult.chk.NumRows() > 0 {
```

Comment:
We may put joinResult.chk back to `joinChkResourceCh` if `joinResult.chk.NumRows() == 0`.

Reply:
We do not need to put it back here since the `innerWorker` will quit if reach this line.

Reply:
Got it.